test/integration/segmentation_camera.cc

```diff
@@ -0,0 +1,262 @@
+
+#include "ignition/rendering/RenderEngine.hh"
+#include "ignition/rendering/RenderingIface.hh"
+#include "ignition/rendering/Scene.hh"
+#include "ignition/rendering/SegmentationCamera.hh"
+
+using namespace ignition;
+using namespace rendering;
+
+//////////////////////////////////////////////////
+class SegmentationCameraTest: public testing::Test,
+  public testing::WithParamInterface<const char *>
+{
+  public: void SegmentationCameraBoxes(const std::string &_renderEngine);
+
+  // Documentation inherited
+  protected: void SetUp() override
+  {
+    ignition::common::Console::SetVerbosity(4);
+  }
+};
+
+/// \brief mutex for thread safety
+std::mutex g_mutex;
+
+/// \brief Segmentation buffer
+uint8_t *g_buffer = nullptr;
+
+/// \brief counter of received segmentation msgs
+int g_counter = 0;
+
+/// \brief callback to get the segmentation buffer
+//////////////////////////////////////////////////
+void OnNewSegmentationFrame(const uint8_t *_data,
+                    unsigned int _width, unsigned int _height,
+                    unsigned int /*_channels*/,
+                    const std::string &/*_format*/)
+{
+  g_mutex.lock();
+  auto bufferSize = _width * _height * 3;
+
+  if (!g_buffer)
+    g_buffer = new uint8_t[bufferSize];
+
+  memcpy(g_buffer, _data, bufferSize);
+
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief Build the scene with 3 boxes besides each other
+/// the 2 aside boxes have the same label & the middle is different
+void BuildScene(rendering::ScenePtr scene)
+{
+  math::Vector3d leftPosition(3, -1.5, 0);
+  math::Vector3d rightPosition(3, 1.5, 0);
+  math::Vector3d middlePosition(3, 0, 0);
+
+  rendering::VisualPtr root = scene->RootVisual();
+
+  // create box visual
+  rendering::VisualPtr box = scene->CreateVisual();
+  box->AddGeometry(scene->CreateBox());
+  box->SetOrigin(0.0, 0.0, 0.0);
+  box->SetLocalPosition(leftPosition);
+  box->SetLocalRotation(0, 0, 0);
+  box->SetUserData("label", 1);
+  root->AddChild(box);
+
+  // create box visual of same label
+  rendering::VisualPtr box1 = scene->CreateVisual();
+  box1->AddGeometry(scene->CreateBox());
+  box1->SetOrigin(0.0, 0.0, 0.0);
+  box1->SetLocalPosition(rightPosition);
+  box1->SetLocalRotation(0, 0, 0);
+  box1->SetUserData("label", 1);
+  root->AddChild(box1);
+
+  // create box visual of different label
+  ignition::rendering::VisualPtr box2 = scene->CreateVisual();
+  box2->AddGeometry(scene->CreateBox());
+  box2->SetOrigin(0.0, 0.0, 0.0);
+  box2->SetLocalPosition(middlePosition);
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", 2);
+  root->AddChild(box2);
+}
+
+//////////////////////////////////////////////////
+void SegmentationCameraTest::SegmentationCameraBoxes(
+  const std::string &_renderEngine)
+{
+  // Optix is not supported
+  if (_renderEngine.compare("optix") == 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' doesn't support depth cameras" << std::endl;
+    return;
+  }
```

Comment:
```suggestion
  // Only ogre2 is supported
  if (_renderEngine.compare("ogre2") != 0)
  {
    igndbg << "Engine '" << _renderEngine
              << "' doesn't support segmentation cameras" << std::endl;
    return;
  }
```
The only engine we currently support for segmentation camera is ogre2, so we should check for this before proceeding